Tensile/Source/lib/include/Tensile/ContractionProblem.hpp

```diff
@@ -505,6 +506,20 @@ namespace Tensile
             return getOperationDescription();
         }
 
+        static constexpr size_t HPA_GSU_WORKSPACE_SIZE_GRANULARITY = 256;
+
+        void setWorkspaceSize(size_t size)
+        {
+            m_workspaceSize = ~size ? (size / HPA_GSU_WORKSPACE_SIZE_GRANULARITY)
```

Comment:
How often is the workspace size going to change?  How often will it change by a value that is <256?

It seems to me it will have a few fairly different values: 0, numeric_limits<size_t>::max(), and D elements * sizeof(float).  It may also later have another larger value if more allocations are done, but the chances that we need to increase the workspace size by some value > 0 and < 256 seems pretty small to me.

This seems to increase complexity at what is not likely to be a meaningful gain.

Reply:
remove this patch.